tasks/ThrusterForce2BodyEffort.cpp

```diff
@@ -0,0 +1,110 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrusterForce2BodyEffort.hpp"
+#include "base/Logging.hpp"
+
+using namespace auv_control;
+
+ThrusterForce2BodyEffort::ThrusterForce2BodyEffort(std::string const& name)
+    : ThrusterForce2BodyEffortBase(name)
+{
+}
+
+ThrusterForce2BodyEffort::ThrusterForce2BodyEffort(std::string const& name, RTT::ExecutionEngine* engine)
+    : ThrusterForce2BodyEffortBase(name, engine)
+{
+}
+
+ThrusterForce2BodyEffort::~ThrusterForce2BodyEffort()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ThrusterForce2BodyEffort.hpp for more detailed
+// documentation about them.
+
+bool ThrusterForce2BodyEffort::configureHook()
+{
+    if (! ThrusterForce2BodyEffortBase::configureHook())
+        return false;
+
+    thrusterMatrix = _thruster_configuration_matrix.get();
+
+    return true;
+}
+bool ThrusterForce2BodyEffort::startHook()
+{
+    if (! ThrusterForce2BodyEffortBase::startHook())
+        return false;
+    return true;
+}
+void ThrusterForce2BodyEffort::updateHook()
+{
+    ThrusterForce2BodyEffortBase::updateHook();
+
+    base::samples::Joints thrusterForces;
+
+    while(_thruster_forces.read(thrusterForces) == RTT::NewData)
+    {
+        base::LinearAngular6DCommand bodyEffort;
+        base::VectorXd thrusterForcesVector;
+        base::Vector6d bodyEffortVector;
+        unsigned int numberOfThrusters = thrusterMatrix.cols();
+        thrusterForcesVector = base::VectorXd::Zero(numberOfThrusters);
+
+        for(int i = 0; i < numberOfThrusters; i++)
```

Comment:
if you anyway touching this, comparing a int with an unsigned int gives you the good old warnings